src/Form/ItemCheckbox.js

```diff
@@ -0,0 +1,116 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import BindMixin from '../Mixin/BindMixin';
+import Util from '../Util/Util';
+
+export default class ItemCheckbox extends Util.mixin(BindMixin) {
+  get methodsToBind() {
+    return ['handleChange'];
+  }
+
+  componentDidMount() {
+    let {props, refs} = this;
+    let checkbox = React.findDOMNode(refs.checkbox);
+
+    if (props.checked != null) {
+      checkbox.checked = props.checked;
+    }
+
+    let indeterminate = props.indeterminate;
+    if (indeterminate != null) {
+      checkbox.indeterminate = indeterminate;
+    }
+  }
+
+  componentDidUpdate() {
+    let {props, refs} = this;
+    let checkbox = React.findDOMNode(refs.checkbox);
```

Comment:
i wonder if we should cache this dom node.

if it's being updated a lot and there are a lot of checkboxes (for example, if you had a checkbox for each row in a table) and that table updates frequently. i wonder if this would be slow or not